src/PollingBlockTracker.ts

```diff
@@ -89,7 +91,19 @@ export class PollingBlockTracker
   async destroy() {
     this._cancelBlockResetTimeout();
     this._maybeEnd();
-    super.removeAllListeners();
+    this.eventNames().forEach((eventName) =>
+      this.listeners(eventName).forEach((listener) => {
+        if (
+          this.#internalEventListeners.every(
+            (internalListener) => !Object.is(internalListener, listener),
+          )
+        ) {
```

Comment:
He did try my suggestion but it resulted in type errors, because the event listener methods provide listers as type `Function` but we track them as a more specific type. We could use a cast or something to fix that, but `Object.is` seemed like a reasonable good solution.

Any suggestions on alternatives that fix the type error?

Reply:
Hmm, perhaps downcasting, but not sure. Not a big deal, though.